### PR TITLE
add filter for opt group

### LINKED ISSUE #561: Separate country and own excerpts
Should countries be displayed separately from own excerpts, even though #558 has made it in?

Alternatively or additionally, should there be a possibility to delete own excerpts?

Alternatively or additionally, should there be an general search page?

Alternatively or additionally, should a filter be displayed? (ie. all, own excerpts only, countries only...)

### PR DESCRIPTION
**Pull request by hixi**
Reviewed by:
- [x] @das-g 

Closes #561

### CODE REVIEW

File: osmaxx/excerptexport/forms/existing_form.py [hunk 1]
```diff
@@ -52,9 +52,22 @@ def __init__(self, *args, **kwargs):
...
+                            </div>
+                        </div>
+                        <div class="col-md-6">
+                            <div id="opt_group_filter_div" class="hidden">
+                                <div class="form-group">
+                                    <label for="opt_group_filter" class="control-label">Filter by Group</label>
+                                    <select class="select form-control" name="opt_group_filter" id="opt_group_filter"></select>
+                                </div>
+                            </div>
+                        </div>
```
Comment (das-g) [inline osmaxx/excerptexport/forms/existing_form.py]: Maybe a good time to pull this out to a separate (HTML partial) file.

Comment (hixi) [inline osmaxx/excerptexport/forms/existing_form.py]: So this would become:
`HTML(render_to_string('excerptexport/forms/partials/filter_layout.html'))`?

Comment (das-g) [inline osmaxx/excerptexport/forms/existing_form.py]: I think so. Unless you prefer to just open an read the file without template processing. (I see no template code in the literal here.)

Comment (hixi) [inline osmaxx/excerptexport/forms/existing_form.py]: I'm in favor of render_to_string, it tells more about the intent and is more djangonistic/djangonautic ;-) 
And if there should be a need to do more than just returning the pure HTML, we only need to change the template (if no context is needed, that is).

Comment (das-g) [inline osmaxx/excerptexport/forms/existing_form.py]: :+1: makes sense

File: osmaxx/excerptexport/static/excerptexport/scripts/excerpt_filter.js [hunk 2]
```diff
@@ -18,75 +18,94 @@
...
+            jQuery("#id_existing_excerpts").children('optgroup').each(
+                    function(count, el){
+                        var label = jQuery(el).attr('label');
+                        jQuery(selectionBox).append('<option value="' + label + '">' + label + '</option>');
+                    }
+            );
+            $('#id_existing_excerpts').filterGroups({groupSelector: '#opt_group_filter'});
+            selectionBox.on('change', function(){
+                wordFilter();
+            });
```
Comment (das-g) [inline osmaxx/excerptexport/static/excerptexport/scripts/excerpt_filter.js]: Simply
```javascript
            selectionBox.on('change', wordFilter);
```
would have the same effect, wouldn't it?

File: osmaxx/excerptexport/static/excerptexport/scripts/excerpt_filter.js [hunk 3]
```diff
@@ -18,75 +18,94 @@
...
-                    }
-                });
-                var selectIEOptions = jQuery('select#id_existing_excerpts > optgroup > span > option');
-                // unwrap matching ones
-                selectIEOptions.filter(':containsCI(' + filterWord + ')').each(function(){
-                    var elem = $(this);
+            // ################# IE patch... #################
+            // Inspired by: http://ajax911.com/hide-options-selecbox-jquery/
+            if (detectIE()) {
+                showAllElements = function () {
```
Comment (das-g) [inline osmaxx/excerptexport/static/excerptexport/scripts/excerpt_filter.js]: Is the space between `function` and `()` intentional?

File: osmaxx/excerptexport/static/excerptexport/scripts/excerpt_filter.js [hunk 4]
```diff
@@ -18,75 +18,94 @@
...
-                            $(opt).show();
-                            $(span).replaceWith(opt);
-                        }
-                    }
-                });
+            var hideElements = function (filterWord) {
+                excerptListFilterFieldClearer.show();
+                excerptListFieldOptions.hide().filter(':containsCI(' + filterWord + ')').show();
+            };
+            var showAllElements = function () {
```
Comment (das-g) [inline osmaxx/excerptexport/static/excerptexport/scripts/excerpt_filter.js]: Is the space between `function` and `()` intentional?

File: osmaxx/excerptexport/static/excerptexport/scripts/excerpt_filter.js [hunk 5]
```diff
@@ -18,75 +18,94 @@
...
-                    if (this.nodeName.toUpperCase() === 'OPTION') {
-                        var span = $(this).parent();
-                        var opt = this;
-                        if ($(this).parent().is('span')) {
-                            $(opt).show();
-                            $(span).replaceWith(opt);
-                        }
-                    }
-                });
+            var hideElements = function (filterWord) {
```
Comment (das-g) [inline osmaxx/excerptexport/static/excerptexport/scripts/excerpt_filter.js]: Is the space between `function` and `()` intentional?

File: osmaxx/excerptexport/static/excerptexport/scripts/excerpt_filter.js [hunk 6]
```diff
@@ -18,75 +18,94 @@
...
-                });
-                var selectIEOptions = jQuery('select#id_existing_excerpts > optgroup > span > option');
-                // unwrap matching ones
-                selectIEOptions.filter(':containsCI(' + filterWord + ')').each(function(){
-                    var elem = $(this);
+            // ################# IE patch... #################
+            // Inspired by: http://ajax911.com/hide-options-selecbox-jquery/
+            if (detectIE()) {
+                showAllElements = function () {
+                    excerptListFieldOptions.each(function (index, val) {
```
Comment (das-g) [inline osmaxx/excerptexport/static/excerptexport/scripts/excerpt_filter.js]: Is the space between `function` and `()` intentional?

File: osmaxx/excerptexport/static/excerptexport/scripts/excerpt_filter.js (hunk 5 above)
Comment (hixi) [inline osmaxx/excerptexport/static/excerptexport/scripts/excerpt_filter.js]: that's unintentional, Pycharm thinks it knows better ...

Comment (das-g) [inline osmaxx/excerptexport/static/excerptexport/scripts/excerpt_filter.js]: > ... and apparently is right: http://javascript.crockford.com/code.html#whitespace and jslint say the same: The word `function` is always followed with one space.

Then we should do that for the newly added code (`optGroupFilter`), too.

Comment (hixi) [inline osmaxx/excerptexport/static/excerptexport/scripts/excerpt_filter.js]: :+1:

**Comment by das-g** [comment]
:+1:

**Pull request merged**